src/balance/balance.service.ts

```diff
@@ -22,28 +23,31 @@ export class BalanceService {
   }
 
   async userLedger(userId: string) {
-    const ledger = await this.paymentRepository.findById(userId);
-    if (ledger.length === 0) {
+    const { paymentId } = await this.ledgerRepository.findById(userId);
+    const payment = await this.paymentRepository.findById(paymentId);
```

Comment:
The names for `payment` and `ledger` need some tweaking.  It's confusing as to what is doing what, at times.